x-pack/plugins/ml/public/application/components/annotations/annotations_table/annotations_table.js

```diff
@@ -447,38 +519,115 @@ export class AnnotationsTable extends Component {
       });
     }
 
-    columns.push({
-      align: RIGHT_ALIGNMENT,
-      width: '60px',
-      name: i18n.translate('xpack.ml.annotationsTable.actionsColumnName', {
-        defaultMessage: 'Actions',
-      }),
-      actions,
-    });
-
     const getRowProps = (item) => {
       return {
         onMouseOver: () => this.onMouseOverRow(item),
         onMouseLeave: () => this.onMouseLeaveRow(),
       };
     };
+    let filterOptions = [];
+    if (this.props.aggregations?.event?.buckets) {
+      const buckets = this.props.aggregations.event.buckets;
+      const foundUser = buckets.findIndex((d) => d.key === 'user') > -1;
+      filterOptions = foundUser ? buckets : [{ key: 'user', doc_count: 0 }, ...buckets];
+    }
+    const filters = [
+      {
+        type: 'field_value_selection',
+        field: 'event',
+        name: 'Event',
+        multiSelect: 'or',
+        options: filterOptions.map((field) => ({
+          value: field.key,
+          name: field.key,
+          view: `${field.key} (${field.doc_count})`,
+        })),
+      },
+    ];
+
+    if (
+      this.props.chartDetails?.entityData?.entities &&
+      this.props.chartDetails?.entityData?.entities.length > 0
+    ) {
+      this.props.chartDetails?.entityData?.entities.forEach((entity) => {
+        if (entity.fieldType === 'partition_field') {
+          columns.push({
+            field: 'partition_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.partitionColumnName', {
+              defaultMessage: 'Partition',
+            }),
+            sortable: true,
+          });
+        }
+
+        if (entity.fieldType === 'over_field') {
+          columns.push({
+            field: 'over_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.overColumnName', {
+              defaultMessage: 'Over',
+            }),
+            sortable: true,
+          });
+        }
+        if (entity.fieldType === 'by_field') {
+          columns.push({
+            field: 'by_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.byColumnName', {
+              defaultMessage: 'By',
+            }),
+            sortable: true,
+          });
+        }
+      });
+
+      filters.push({
+        type: 'is',
+        field: 'current_series',
+        name: 'Current Series',
```

Comment:
Can this text be internationalized? Also I can't decide on what works best for the text! `Filter for series`? `Series only`. Whatever the text, we should use sentence case here (so 'Current series' if sticking with this label).

I guess unless we make this a custom search control (like a checkbox) there is no easy way to indicate what state the selection is in (Series only, or Series plus job) ?

Reply:
![image](https://user-images.githubusercontent.com/3756330/86272005-e84e6c00-bb9b-11ea-81d7-64ff8a86a8e9.png)
Not sure I know all the context, but do you just have a filter group with `Series` and `Series plus Job`.  Might get quite long with internationalization, so maybe the 'Filter for series' like you mentioned and have a dropdown for 'Series only' and 'Series plug job'
http://eui.elastic.co/#/forms/filter-group

Reply:
Will try that out!

Reply:
Added i18n translations here https://github.com/elastic/kibana/pull/70198/commits/55aa5c83034c618d1e8c962093bca83a25abcbfb